src/parser/jobs/blm/modules/Gauge.tsx

```diff
@@ -40,99 +49,135 @@ export interface BLMGaugeState {
-	private targetedIceSpellIds = ICE_SPELLS_TARGETED.map(key => this.data.actions[key].id)
-	private untargetedIceSpellIds = ICE_SPELLS_UNTARGETED.map(key => this.data.actions[key].id)
 	private iceSpellIds = [
-		...this.targetedIceSpellIds,
-		...this.untargetedIceSpellIds,
-	]
-
-	private affectsGaugeOnDamage: number[] = [
-		...this.fireSpellIds,
-		...this.targetedIceSpellIds,
-	]
-	private affectsGaugeOnCast: number[] = [
-		...this.untargetedIceSpellIds,
-		this.data.actions.TRANSPOSE.id,
-		this.data.actions.ENOCHIAN.id,
-		this.data.actions.FOUL.id,
-		this.data.actions.XENOGLOSSY.id,
+		...ICE_SPELLS_TARGETED.map(key => this.data.actions[key].id),
+		...ICE_SPELLS_UNTARGETED.map(key => this.data.actions[key].id),
 	]
+	private affectsGaugeOnDamage = AFFECTS_GAUGE_ON_DAMAGE.map(key => this.data.actions[key].id)
 
 	private castTimeIndex: number | null = null
 
+	/** Astral Fire */
+	private astralFireGauge = this.add(new CounterGauge({
+		maximum: MAX_ASTRAL_UMBRAL_STACKS,
+		graph: {
+			handle: ASTRAL_UMBRAL_HANDLE,
+			label: <Trans id="blm.gauge.resource.astral-fire">Astral Fire</Trans>,
+			color: FIRE_COLOR.fade(STANCE_FADE),
+		},
+	}))
+	private astralFireTimer = this.add(new TimerGauge({
+		maximum: ASTRAL_UMBRAL_DURATION,
+		onExpiration: this.onAstralUmbralTimeout.bind(this),
+		graph: {
+			handle: ASTRAL_UMBRAL_HANDLE,
+			label: <Trans id="blm.gauge.resource.astral-timer">Astral Fire Timer</Trans>,
+			color: FIRE_COLOR.fade(TIMER_FADE),
+		},
+	}))
+	/** Umbral Ice */
+	private umbralIceGauge = this.add(new CounterGauge({
+		maximum: MAX_ASTRAL_UMBRAL_STACKS,
+		graph: {
+			handle: ASTRAL_UMBRAL_HANDLE,
+			label: <Trans id="blm.gauge.resource.umbral-ice">Umbral Ice</Trans>,
+			color: ICE_COLOR.fade(STANCE_FADE),
+		},
+	}))
+	private umbralIceTimer = this.add(new TimerGauge({
+		maximum: ASTRAL_UMBRAL_DURATION,
+		onExpiration: this.onAstralUmbralTimeout.bind(this),
+		graph: {
+			handle: ASTRAL_UMBRAL_HANDLE,
+			label: <Trans id="blm.gauge.resource.umbral-timer">Umbral Ice Timer</Trans>,
+			color: ICE_COLOR.fade(TIMER_FADE),
+		},
+	}))
+
+	/** Umbral Hearts */
+	private umbralHeartsGauge = this.add(new CounterGauge({
+		maximum: MAX_UMBRAL_HEART_STACKS,
+		graph: {
+			label: <Trans id="blm.gauge.resource.umbral-hearts">Umbral Hearts</Trans>,
+			color: ICE_COLOR.fade(GAUGE_FADE),
+		},
+	}))
+
+	/** Polyglot */
+	private polyglotGauge = this.add(new CounterGauge({
+		maximum: MAX_POLYGLOT_STACKS,
+		graph: {
+			label: <Trans id="blm.gauge.resource.polyglot">Polyglot</Trans>,
+			color: POLYGLOT_COLOR.fade(GAUGE_FADE),
+		},
+	}))
+	private polyglotTimer = this.add(new TimerGauge({
+		maximum: ENOCHIAN_DURATION_REQUIRED,
+		onExpiration: this.onGainPolyglot.bind(this),
+		graph: {
+			label: <Trans id="blm.gauge.resource.polyglot-timer">Polyglot Timer</Trans>,
+			color: POLYGLOT_COLOR.fade(TIMER_FADE),
+		},
+	}))
+	private enochianGauge = this.add(new CounterGauge({
+		maximum: 1,
+	}))
```

Comment:
I'd likely just track this as a boolean rather than gauge. It's either on or off.